src/IO/Config.hs

```diff
@@ -39,7 +40,19 @@ defaultConfig = Config {
 }
 
 getDir :: IO FilePath
-getDir = (++ "/.taskell") <$> getHomeDirectory
+getDir =
+    let
+        directoryName = "taskell"
+        oldConfigDir = (++ "/." ++ directoryName) <$> getHomeDirectory
+        xdgDefaultConfigDirectory = (++ "/.config/" ++ directoryName) <$> getHomeDirectory
+        iOMaybeXdgEnvironmentConfigDirectory = lookupEnv "XDG_CONFIG_HOME"
+        xdgDirectory = do
+            maybeXdgEnvironmentConfigDirectory <- iOMaybeXdgEnvironmentConfigDirectory
+            case maybeXdgEnvironmentConfigDirectory  of
+                 Just xdgEnvironmentConfigDirectory -> pure $ xdgEnvironmentConfigDirectory ++ "/" ++ directoryName
+                 Nothing -> xdgDefaultConfigDirectory
+    in
+    oldConfigDir >>= doesDirectoryExist >>= (\oldconfigAvailable -> if oldconfigAvailable then oldConfigDir else xdgDirectory)
```

Comment:
Now that you are in the `do` block, you don't need to use `>>=` anymore. This becomes the more readable:

```
  home <-getHomeDirectory
  let oldConfigPath = home ++ "'/." ++ directoryName
  oldConfigAvailable <- doesDirectoryExist oldConfigPath

  if oldConfigAvailable
    then pure oldConfigPath
    else do
      xdgEnv = lookupEnv "XDG_CONFIG_HOME"
      pure $ case xdgEnv  of
        Just xdgConfigHome -> xdgConfigHome ++ "/" ++ directoryName
        Nothing -> home ++ "/.config/" ++ directoryName
```

Reply:
Thank you, I've somehow thought that `do` blocks wouldn't be considered good practice but it seems I'm wrong. I will provide a fix soonish, thanks.

Reply:
Some people don't like `do` blocks as they hide what's really going on, but they're definitely preferable to using `let` to store monadic values. You could remove the `do` completely by breaking it up into smaller functions, but probably not necessary.

Reply:
Currently try to refactor into smaller funcs :)